tests/unit/spawn_test.cc

```diff
@@ -0,0 +1,82 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2022 Kefu Chai ( tchaikov@gmail.com )
+ */
+#include <seastar/core/future.hh>
+#include <seastar/core/thread.hh>
+#include <seastar/util/process.hh>
+#include <seastar/testing/test_case.hh>
+#include <seastar/testing/thread_test_case.hh>
+#include <string>
+
+using namespace seastar;
+using namespace seastar::experimental;
+
+SEASTAR_TEST_CASE(test_spawn_success) {
+    const char* true_cmd = "/usr/bin/true";
+    return spawn_process(true_cmd, {true_cmd}).then([](auto process) {
+        return process.wait();
+    }).then([] (auto wstatus) {
+        auto* exit_status = std::get_if<process::exit_status>(&wstatus);
+        BOOST_REQUIRE(exit_status != nullptr);
+        BOOST_CHECK_EQUAL(exit_status->value, EXIT_SUCCESS);
+    });
+}
+
+SEASTAR_TEST_CASE(test_spawn_failure) {
+    const char* false_cmd = "/usr/bin/false";
+    return spawn_process(false_cmd, {false_cmd}).then([](auto process) {
+        return process.wait();
+    }).then([] (auto wstatus) {
+        auto* exit_status = std::get_if<process::exit_status>(&wstatus);
+        BOOST_REQUIRE(exit_status != nullptr);
+        BOOST_CHECK_EQUAL(exit_status->value, EXIT_FAILURE);
+    });
+}
+
+SEASTAR_TEST_CASE(test_spawn_echo) {
+    const char* echo_cmd = "/usr/bin/echo";
+    return spawn_process(echo_cmd, {echo_cmd, "-n", "hello", "world"}).then([](auto process) {
+        return do_with(std::move(process), bool(true), [](auto& p, auto& matched) {
+            using consumption_result_type = typename input_stream<char>::consumption_result_type;
+            using stop_consuming_type = typename consumption_result_type::stop_consuming_type;
+            using tmp_buf = stop_consuming_type::tmp_buf;
+            struct consumer {
+                consumer(std::string_view expected, bool& matched)
+                    : _expected(expected), _matched(matched) {}
+                future<consumption_result_type> operator()(tmp_buf buf) {
+                    _matched = std::equal(buf.begin(), buf.end(), _expected.begin());
+                    if (!_matched) {
+                        return make_ready_future<consumption_result_type>(stop_consuming_type({}));
+                    }
+                    _expected.remove_prefix(buf.size());
+                    return make_ready_future<consumption_result_type>(continue_consuming{});
+                }
+                std::string_view _expected;
+                bool& _matched;
+            };
+            return p.stdout().consume(consumer("hello world", matched)).then([&matched] {
+                BOOST_CHECK(matched);
+            }).finally([&p] {
+                return p.wait().discard_result();
+            });
+        });
+    });
```

Comment:
Please add a case for /non/existent/path.

Reply:
sure. added.